internal/services/services.go

```diff
@@ -0,0 +1,486 @@
+package services
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/chef/omnitruck-service/clients"
+	"github.com/chef/omnitruck-service/clients/omnitruck"
+	"github.com/chef/omnitruck-service/clients/omnitruck/replicated"
+	"github.com/chef/omnitruck-service/config"
+	"github.com/chef/omnitruck-service/constants"
+	"github.com/chef/omnitruck-service/dboperations"
+	helpers "github.com/chef/omnitruck-service/internal/helper"
+	"github.com/chef/omnitruck-service/internal/strategy"
+	"github.com/chef/omnitruck-service/logger"
+	"github.com/chef/omnitruck-service/utils/template"
+	"github.com/gofiber/fiber/v2"
+	"github.com/samber/do"
+	log "github.com/sirupsen/logrus"
+)
+
+type DownloadService struct {
+	Log               *log.Entry
+	Validator         omnitruck.RequestValidator
+	DatabaseService   dboperations.IDbOperations
+	TemplateRenderer  template.TemplateRender
+	Replicated        replicated.IReplicated
+	LicenseClient     clients.ILicense
+	LicenseServiceUrl string
+	Mode              constants.ApiType
+	locals            map[string]interface{}
+}
+
+func NewDownloadService(c *fiber.Ctx, log *log.Entry) (*DownloadService, error) {
+	service := DownloadService{}
+	service.SetLocals(c)
+	service.Log = log
+	// Retrieve the injector from the Fiber context
+	reqInjectorI := c.Locals("reqinjector")
+	reqInjector, ok := reqInjectorI.(*do.Injector)
+	if !ok {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, "Failed to retrieve request injector")
+	}
+
+	validator := do.MustInvokeNamed[omnitruck.RequestValidator](reqInjector, "validator")
```

Comment:
I think this setting of service variables should happen in `ProductStrategyDeps` func as no where else it is used in code. Also this would mean that if anybody wants to use these they have to use strategy and not directly use it in there service methods

Reply:
Like discussed, will take up these as a separate PR tracked under same story. So that this PR can be merged and unblock other development items